### PR TITLE
CDAP-15640 GCS integration tests coverage

### PR DESCRIPTION
**Pull request by vladhlinskiy**
Added tests for GCS Move, Delete and Create actions.

### CODE REVIEW

**Comment by vladhlinskiy** [comment]
@albertshau could you, please, take a look?

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java [hunk 1]
```diff
@@ -193,7 +212,7 @@ public void testGCSCopy() throws Exception {
         cp bucket1/catalog.txt bucket4/dir1 when 'dir1' already exists
         should copy the file into bucket4/dir1/catalog.txt
      */
-    bucket4.create("dir1/", new byte[] { });
```
Comment (albertshau) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: don't need to make these changes

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java [hunk 2]
```diff
@@ -70,8 +70,14 @@
...
   private static final String INPUT_BLOB_NAME = "data/input/customers.csv";
   private static final String OUTPUT_BLOB_NAME = "data/output";
+  private static final String GCS_BUCKET_DELETE_PLUGIN_NAME = "GCSBucketDelete";
+  private static final String GCS_BUCKET_CREATE_PLUGIN_NAME = "GCSBucketCreate";
+  private static final String GCS_MOVE_PLUGIN_NAME = "GCSMove";
+  private static final String GCS_COPY_PLUGIN_NAME = "GCSCopy";
+  private static final String FILE_PLUGIN_PLUGIN_NAME = "File";
   private static Storage storage;
   private List<Bucket> createdBuckets;
+  private List<String> markedForDeleteBuckets;
```
Comment (albertshau) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java:80]: lets combine this with createdBuckets

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java [hunk 3]
```diff
@@ -108,6 +115,14 @@ protected void innerTearDown() {
         LOG.error("Unable to delete GCS bucket {}", bucket.getName(), e);
       }
     }
+    for (String bucketName : markedForDeleteBuckets) {
+      try {
+        Bucket bucket = storage.get(bucketName);
```
Comment (albertshau) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java:112]: if bucket is null, should skip the rest of the delete logic.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java [hunk 4]
```diff
@@ -297,14 +316,183 @@ public void testGCSCopy() throws Exception {
...
+    bucket1.create("dir/1/2.txt", "1-2".getBytes(StandardCharsets.UTF_8));
+    bucket1.create("dir/2/1.txt", "2-1".getBytes(StandardCharsets.UTF_8));
+    bucket1.create("dir/2/2.txt", "2-2".getBytes(StandardCharsets.UTF_8));
+    bucket1.create("1.txt", "1".getBytes(StandardCharsets.UTF_8));
+    bucket1.create("2.txt", "2".getBytes(StandardCharsets.UTF_8));
+
+    ETLStage cp1 = createMoveStage("mv1", bucket1Name, bucket2Name, false);
+
+    // deploy the pipeline
+    ETLBatchConfig config = ETLBatchConfig.builder("* * * * *")
```
Comment (albertshau) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: this is a deprecated method. should used `ETLBatchConfig.builder()`

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java [hunk 5]
```diff
@@ -329,15 +517,15 @@ public void testGCSToGCS() throws Exception {
     String sourceSchema = "{\"type\":\"record\",\"name\":\"etlSchemaBody\"," +
       "\"fields\":[{\"name\":\"offset\",\"type\":\"long\"},{\"name\":\"body\",\"type\":\"string\"}]}";
     ETLStage source = new ETLStage("GCSSourceStage",
-                                   new ETLPlugin("File",
+                                   new ETLPlugin(FILE_PLUGIN_PLUGIN_NAME,
```
Comment (albertshau) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: this pipeline is actually using the 'File' plugins, which are different than the 'GCS' plugins. Can you change them to use 'GCS'? I don't think any of the config properties need to change.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java (hunk 1 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: Thanks. Fixed.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java (hunk 2 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java:80]: Thanks. Fixed.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java (hunk 3 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java:112]: Thanks. Fixed.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java (hunk 4 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: Thanks. Fixed.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java (hunk 5 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: Thanks. Fixed.

**Comment by vladhlinskiy** [comment]
@albertshau could you, please, take a look? All comments were addressed.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java [hunk 6]
```diff
@@ -96,16 +102,19 @@ protected void innerSetup() throws Exception {
...
   }
 
   @Override
   protected void innerTearDown() {
-    for (Bucket bucket : createdBuckets) {
+    for (String bucketName : markedForDeleteBuckets) {
       try {
-        deleteBucket(bucket);
+        Bucket bucket = storage.get(bucketName);
+        if(bucket != null) {
```
Comment (albertshau) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: missing space after 'if'

Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/GCSTest.java]: Done.

**Comment by vladhlinskiy** [comment]
@albertshau could you, please, merge?

**Pull request merged**